### PR TITLE
add docstrings for domains, models, and simulation modules

### LINKED ISSUE #82: Provide comprehensive docstrings for Domain, Models, and Simulation modules
**context**
Now that the documenter has been set up, we can add some more comprehensive docs.

**ideas**
I think we want to keep documentation minimal but sufficient at this stage (to not have too much maintenance). I propose that we follow [these](https://docs.julialang.org/en/v1/manual/documentation/#man-documentation) conventions. The goals are to have documentation come in through doc strings primarily so that the documentation can be easily reviewed in PRs.

**next steps**
Compare the proposed documentation guidelines with the old ones in `CM.jl` and `O.jl` and add an example or link to the contributor guide and also set up the first modules in the finalized style.

### PR DESCRIPTION
**Pull request by szy21**
Co-authored-by: @akshaysridhar 
Closes #82

### CODE REVIEW

File: src/Simulations/simulation.jl [hunk 1]
```diff
@@ -38,7 +35,13 @@ function Simulation(model::AbstractModel, method; Y_init = nothing, dt, tspan)
...
 
 """
-    set!
+    set!(
+        simulation::AbstractSimulation,
+        submodel_name = nothing;
+        kwargs...,
+    )
+
+Set up the `simulation`.
```
Comment (bischtob) [inline src/Simulations/simulation.jl:44]: This sets the simulation state to a new state, either through an array or a function. This is signified by the "!" after the "set", aka it updates the existing simulation instance that is passed as first argument.

**Review by bischtob — APPROVED** [review]
There is only one clarification :), so this is great work! Please update the docstring and squash & rebase. Then feel free to r+!

File: src/Simulations/simulation.jl [hunk 2]
```diff
@@ -96,8 +99,18 @@ function set!(
     nothing
 end
 
+"""
+    step!(sim::AbstractSimulation, args...; kwargs...)
+
+Step forward a simulation `sim` one time step.
+"""
 step!(sim::AbstractSimulation, args...; kwargs...) =
```
Comment (szy21) [inline src/Simulations/simulation.jl:107]: @bischtob Could I change `sim` here to `simulation` to be consistent with the `set!` function?

Comment (bischtob) [reply src/Simulations/simulation.jl:107]: Sounds good!

**Review by akshaysridhar — APPROVED** [review]
LGTM as a first step towards doc building. I like examples in the Oceananigans style which we can work towards in future PRs

**Pull request merged**